libcaf_core/src/node_id.cpp

```diff
@@ -64,12 +65,24 @@ node_id node_id::default_data::local(const actor_system_config&) {
   macs.reserve(ifs.size());
   for (auto& i : ifs)
     macs.emplace_back(std::move(i.second));
-  auto hd_serial_and_mac_addr = join(macs, "") + detail::get_root_uuid();
+  auto seeded_hd_serial_and_mac_addr = join(macs, "") + detail::get_root_uuid();
```

Comment:
Why not use RFC 4122 to generate a UUID?

Reply:
At this point, we might as well. Originally, producing the same `node_id` for each run was intended as feature. 😉

However, switching to UUIDs in node IDs would potentially break downstream code. CAF uses 160-bit numbers, while UUIDs are 128-bit numbers. Since `caf_io` is already on its way out, I rather not change data formats or APIs at this point.